File: AlgorithmsPackPythonSolutions/Eliminate Maximum Number of Monsters.py
class Solution:
    def eliminateMaximum(self, dist: List[int], speed: List[int]) -> int:
        sz = len(dist)
        arrivalTime = [(d + s - 1) // s for d, s in zip(dist, speed)]
        arrivalTime.sort()
        
        monster = 0
        for i in range(len(arrivalTime)):
            if i >= arrivalTime[monster]:
                break
            monster += 1
        
        return monster

File: AlgorithmsPackCPP/Eliminate Maximum Number of Monsters.cpp
class Solution {
public:
    int eliminateMaximum(vector<int>& dist, vector<int>& speed) {
        int sz = dist.size();
        vector<int>arrivalTime(sz);
        for(int i = 0; i < sz; i++){
            if(dist[i]%speed[i]==0)
                arrivalTime[i] = dist[i]/speed[i];
            else
                arrivalTime[i] = dist[i]/speed[i]+1;
        }
        sort(arrivalTime.begin(),arrivalTime.end());
        
        int monster = 0;
        for(int i = 0; i < arrivalTime.size() && i < arrivalTime[monster]; i++)
            monster++;
        
        return monster;
    }
};
